crates/sui-rpc-loadgen/src/payload/rpc_command_processor.rs

```diff
@@ -126,6 +135,38 @@ impl RpcCommandProcessor {
             }
         }
     }
+
+    pub(crate) fn add_transaction_digests(&self, digests: Vec<TransactionDigest>) {
+        // extend method requires mutable access to the underlying DashSet, which is not allowed by the Arc
+        for digest in digests {
+            self.transaction_digests.insert(digest);
+        }
+    }
+
+    pub(crate) fn add_addresses(&self, responses: Vec<SuiTransactionResponse>) {
+        for response in responses {
+            let transaction = response.transaction;
+            if let Some(transaction) = transaction {
+                let data = transaction.data;
+                match data {
+                    SuiTransactionData::V1(v1) => {
+                        self.addresses.insert(v1.sender);
+                    }
+                }
+            }
+        }
+    }
+
+    pub(crate) fn dump_cache_to_file(&self) {
+        let data_dir = "~/.sui/sui_config/data";
```

Comment:
sorry my comment in https://github.com/MystenLabs/sui/pull/9874#discussion_r1148172429 might be confusing - we should still put this in `Opts`, but the default path should be `~/.sui/load_gen` instead of `~/.sui/sui_config` because what we store is not a config